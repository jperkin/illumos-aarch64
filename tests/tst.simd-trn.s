/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2017 Joyent, Inc.
 */

/*
 * C3.6.3 AdvSIMD ZIP/UZP/TRN
 */
.macro simd_trn
	.irp op,uzp1,trn1,zip1,uzp2,trn2,zip2
	.irp arr,8b,16b,4h,8h,2s,4s,2d
	\op	v0.\arr, v15.\arr, v31.\arr
	.endr
	.endr
.endm

.text
.globl libdis_test
.type libdis_test, @function
libdis_test:
	simd_trn
.size libdis_test, [.-libdis_test]
