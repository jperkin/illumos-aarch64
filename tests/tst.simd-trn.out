    libdis_test:      e0 19 1f 0e  uzp1 v0.8b, v15.8b, v31.8b
    libdis_test+0x4:  e0 19 1f 4e  uzp1 v0.16b, v15.16b, v31.16b
    libdis_test+0x8:  e0 19 5f 0e  uzp1 v0.4h, v15.4h, v31.4h
    libdis_test+0xc:  e0 19 5f 4e  uzp1 v0.8h, v15.8h, v31.8h
    libdis_test+0x10: e0 19 9f 0e  uzp1 v0.2s, v15.2s, v31.2s
    libdis_test+0x14: e0 19 9f 4e  uzp1 v0.4s, v15.4s, v31.4s
    libdis_test+0x18: e0 19 df 4e  uzp1 v0.2d, v15.2d, v31.2d
    libdis_test+0x1c: e0 29 1f 0e  trn1 v0.8b, v15.8b, v31.8b
    libdis_test+0x20: e0 29 1f 4e  trn1 v0.16b, v15.16b, v31.16b
    libdis_test+0x24: e0 29 5f 0e  trn1 v0.4h, v15.4h, v31.4h
    libdis_test+0x28: e0 29 5f 4e  trn1 v0.8h, v15.8h, v31.8h
    libdis_test+0x2c: e0 29 9f 0e  trn1 v0.2s, v15.2s, v31.2s
    libdis_test+0x30: e0 29 9f 4e  trn1 v0.4s, v15.4s, v31.4s
    libdis_test+0x34: e0 29 df 4e  trn1 v0.2d, v15.2d, v31.2d
    libdis_test+0x38: e0 39 1f 0e  zip1 v0.8b, v15.8b, v31.8b
    libdis_test+0x3c: e0 39 1f 4e  zip1 v0.16b, v15.16b, v31.16b
    libdis_test+0x40: e0 39 5f 0e  zip1 v0.4h, v15.4h, v31.4h
    libdis_test+0x44: e0 39 5f 4e  zip1 v0.8h, v15.8h, v31.8h
    libdis_test+0x48: e0 39 9f 0e  zip1 v0.2s, v15.2s, v31.2s
    libdis_test+0x4c: e0 39 9f 4e  zip1 v0.4s, v15.4s, v31.4s
    libdis_test+0x50: e0 39 df 4e  zip1 v0.2d, v15.2d, v31.2d
    libdis_test+0x54: e0 59 1f 0e  uzp2 v0.8b, v15.8b, v31.8b
    libdis_test+0x58: e0 59 1f 4e  uzp2 v0.16b, v15.16b, v31.16b
    libdis_test+0x5c: e0 59 5f 0e  uzp2 v0.4h, v15.4h, v31.4h
    libdis_test+0x60: e0 59 5f 4e  uzp2 v0.8h, v15.8h, v31.8h
    libdis_test+0x64: e0 59 9f 0e  uzp2 v0.2s, v15.2s, v31.2s
    libdis_test+0x68: e0 59 9f 4e  uzp2 v0.4s, v15.4s, v31.4s
    libdis_test+0x6c: e0 59 df 4e  uzp2 v0.2d, v15.2d, v31.2d
    libdis_test+0x70: e0 69 1f 0e  trn2 v0.8b, v15.8b, v31.8b
    libdis_test+0x74: e0 69 1f 4e  trn2 v0.16b, v15.16b, v31.16b
    libdis_test+0x78: e0 69 5f 0e  trn2 v0.4h, v15.4h, v31.4h
    libdis_test+0x7c: e0 69 5f 4e  trn2 v0.8h, v15.8h, v31.8h
    libdis_test+0x80: e0 69 9f 0e  trn2 v0.2s, v15.2s, v31.2s
    libdis_test+0x84: e0 69 9f 4e  trn2 v0.4s, v15.4s, v31.4s
    libdis_test+0x88: e0 69 df 4e  trn2 v0.2d, v15.2d, v31.2d
    libdis_test+0x8c: e0 79 1f 0e  zip2 v0.8b, v15.8b, v31.8b
    libdis_test+0x90: e0 79 1f 4e  zip2 v0.16b, v15.16b, v31.16b
    libdis_test+0x94: e0 79 5f 0e  zip2 v0.4h, v15.4h, v31.4h
    libdis_test+0x98: e0 79 5f 4e  zip2 v0.8h, v15.8h, v31.8h
    libdis_test+0x9c: e0 79 9f 0e  zip2 v0.2s, v15.2s, v31.2s
    libdis_test+0xa0: e0 79 9f 4e  zip2 v0.4s, v15.4s, v31.4s
    libdis_test+0xa4: e0 79 df 4e  zip2 v0.2d, v15.2d, v31.2d
